install/nodejs

```diff
@@ -0,0 +1,244 @@
+    {
+        edebug "$(cat "${sha_output_path}")"
+        die "Failed to retrieve version from ${sha_output_path}"
+    }
+}
+
+opt_usage install_node <<'END'
+Installs the nodejs lts version based on the codename passed in
+END
+install_nodejs()
+{
+    $(opt_parse                                                           \
+        ":install_prefix p | The install prefix"                          \
+        ":lts_codename   c | The codename for the lts version to install"
+    )
+
+    lts_version=$(get_latest_node_version --lts_codename "${nodejs_codename}" --sha_output_path "${default_sha_path}")
+    einfo "Installing latest version of nodejs codename '${lts_codename}': ${lts_version}"
+
+    if os darwin; then
+        package=""
+
+        mac_lts_version=$(cut -d. -f1 <<< "${lts_version}")
+        package="node@${mac_lts_version}"
+        edebug "Mac LTS package to install: ${package}"
+
+        # Install the latest codenamed version of node using brew
+        pkg_install --sync "node@${mac_lts_version}"
+
+    # If the distro is linux, always download the lts codename's latest versioned binary
+    elif os linux; then
+
+        # Download tarball
+        lts_url="https://nodejs.org/download/release/latest-${lts_codename}/node-v${lts_version}-linux-x64.tar.gz"
+        tarball="${TMPDIR:-/tmp}/$(basename "${lts_url}")"
+
+        edebug "$(lval lts_version tarball lts-url)"
+
+        eretry efetch --style=einfo "${lts_url}" "${tarball}"
+        trap_add "rm -f ${tarball}"
+
+        # Extract tarball
+        mkdir -p "${install_prefix}/bin" || die "Failed creating ${install_prefix}/bin, check for correct permissions"
+        tar --directory "${install_prefix}" --extract --gzip --file "${tarball}"
+        nodejs_dir="${install_prefix}/node-v${lts_version}-linux-x64"
+
+        # Clean up symlinks
+        cleanup_node_symlinks --install_prefix "${install_prefix}"
+
+        # Create the high level symlink
+        lts_location="${install_prefix}/node-v${lts_version}-linux-x64"
+        einfo "Creating symlink to ${lts_location}"
+        ln -sv "${lts_location}" "${install_prefix}/node"
+
+        # Add symlinks to <install_prefix>/bin for all node binaries
+        einfo "Creating symbolic links for node binaries"
+        ln -sv "${install_prefix}/node/bin/"* "${install_prefix}/bin"
+    fi
+
+    # Verify installation
+    verify_symlinks --install_prefix "${install_prefix}" --lts_version "${lts_version}"
+}
+
+opt_usage verify_symlinks <<'END'
+Verify the symlinks were properly created based on the entries in <prefix>/node/bin
+END
+verify_symlinks()
+{
+    $(opt_parse                                                         \
+        ":install_prefix p | The install prefix"                        \
+        ":lts_version    v | The lts version to install"                \
+    )
+
+    ## Verify binaries provided by the lts package are in the path
+    nodejs_dir="${install_prefix}/node-v${lts_version}-linux-x64/bin"
```

Comment:
does Mac os also follow this install directory structure?

Reply:
That is an excellent point and great catch! No, it doesn't, I've moved the verify symlinks to under the linux check in the install function. Mac OS install will be controlled by brew.